lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -1307,6 +1308,63 @@ def upload_exec_summary_report(_xmodule_instance_args, _entry_id, course_id, _ta
     return task_progress.update_task_state(extra_meta=current_step)
 
 
+def upload_course_survey_report(_xmodule_instance_args, _entry_id, course_id, _task_input, action_name):  # pylint: disable=invalid-name
+    """
+    For a given `course_id`, generate a html report containing the survey results for a course.
+    """
+    start_time = time()
+    start_date = datetime.now(UTC)
+    num_reports = 1
+    task_progress = TaskProgress(action_name, num_reports, start_time)
+
+    current_step = {'step': 'Gathering course survey report information'}
+    task_progress.update_task_state(extra_meta=current_step)
+
+    distinct_survey_fields_queryset = SurveyAnswer.objects.values('field_name').distinct()
+    survey_fields = []
+    for unique_field_row in distinct_survey_fields_queryset:
+        survey_fields.append(unique_field_row['field_name'])
+    survey_fields.sort()
+
+    user_survey_answers = OrderedDict()
+    survey_answers_for_course = SurveyAnswer.objects.filter(course_key=course_id)
+
+    for survey_field_record in survey_answers_for_course:
+        user_id = survey_field_record.user.id
+        if user_id in user_survey_answers.keys():
+            user_survey_answers[user_id]['field_name'] = survey_field_record.field_value
```

Comment:
I think the dictionary key needs to be survey_field_record.field_name

so, user_survey_answers[user_id][field_name] = survey_field_record.field_value